lib/elixir/lib/module/checker.ex

```diff
@@ -47,6 +47,14 @@ defmodule Module.Checker do
     |> emit_warnings()
   end
 
+  defp no_warn_undefined(module_map) do
+    for(
+      {:no_warn_undefined, values} <- module_map.compile_opts,
+      value <- List.wrap(values),
+      do: value
+    ) ++ Code.compiler_options()[:no_warn_undefined]
+  end
```

Comment:
By doing this we are doing some premature work for something we may never use. My suggestion is to compute this on `no_warn_undefined?`.

Reply:
OTOH we will have to recompute it for every warning and skipped warning. Right now we are only precomputing it once per module.

Reply:
Good point. Sounds good to me then.